scipy/stats/stats.py

```diff
@@ -3999,6 +3996,51 @@ def pearsonr(x, y):
     >>> stats.pearsonr([1, 2, 3, 4, 5], [10, 9, 2.5, 6, 4])
     (-0.7426106572325057, 0.1505558088534455)
 
+    It is important to keep in mind that no correlation does not imply
+    independence unless (x, y) is Gaussian. Correlation can even be zero
+    when there is a very simple dependence structure: if x follows a
+    standard normal distribution, let y = abs(x). A simple check
+    confirms that the correlation is close to zero:
+
+    >>> x = stats.norm.rvs(size=500)
+    >>> y = np.abs(x)
+    >>> stats.pearsonr(x, y)
+    (-0.016172891856853524, 0.7182823678751942) # may vary
+
+    Indeed, since the expectation of x is zero, cov(x, y) = E[x*y]. This equals
+    E[x*abs(x)] which is zero by symmetry.
+
+    If the relationship between x and y is non-linear, the correlation
+    coefficient can be misleading. For example, if x has a standard normal
+    distribution, define y = x if x < 0 and y = 0 otherwise. A simple
+    alculation shows that corr(x, y) = sqrt(2/Pi) = 0.797..., implying a
+    high level of correlation:
+
+    >>> y = np.where(x < 0, x, 0)
+    >>> stats.pearsonr(x, y)
+    (0.8537091583771509, 3.183461621422181e-143) # may vary
+
+    This is unintuitive since there is no dependence of x and y if x is larger
+    than zero which happens in about half of the cases if we sample x and y.
+
+    There is a linear dependance between x and y if y = a + b*x + e, where
+    a,b are constants and e is a random error term, assumed to be independent
+    of x. For simplicity, assume that x is standard normal, a=0, b=1 and let
+    e follow a normal distribution with mean zero and standard deviation s>0.
+
+    >>> s = 0.5
+    >>> e = stats.norm.rvs(scale=s, size=500)
+    >>> y = x + e
+    >>> stats.pearsonr(x, y)
+    (0.9029601878969703, 8.428978827629898e-185) # may vary
+
+    This should be close to the exact value given by
+
+    >>> 1/np.sqrt(1 + s**2)
+    0.8944271909999159
+
+    As expected, a large variance of the noise reduces the correlation, while
+    the correlation approaches one as the variance of the error goes to zero.
```

Comment:
Is `s = 0.5` supposed to be "a large variance"? "As expected" suggests that the numerical example is an illustration of the statement, but when I was reading I was confused because `s = 0.5` is a large variance. 

Maybe put this part before the "bad" examples because this is the way the correlation coefficient is _supposed_ to work.

Reply:
I moved the example up and also pointed out that the correlation is high for s=0.5